xbmc/utils/UnicodeConverter.cpp

```diff
@@ -0,0 +1,992 @@
+
+static constexpr size_t paddingBytes{512};
+
+class CUnicodeConverterType
+{
+  //
+  // Since this is based on CharSetConverter there are some artifacts that are not
+  // so necessary now, however, they are kept to help any effort to merge any of
+  // the changes here into CharSetConverter. Much was changed, but the skeleton is
+  // still present. The CUnicodeConverterType is one of the items that is not
+  // so necessary with UnicodeConverter.
+  //
+  // Simple class to keep the necessary information to create an iconv instance
+  // as well as that instance once created.
+
+public:
+  CUnicodeConverterType(const std::string_view sourceCharset,
+                        const std::string_view targetCharset,
+                        size_t targetSingleCharMaxLen = 1);
+  ~CUnicodeConverterType();
+
+  iconv_t PrepareConverter();
+  CUnicodeConverterType& operator=(const CUnicodeConverterType& other)
+  {
+    m_sourceCharset = other.m_sourceCharset;
+    m_targetCharset = other.m_targetCharset;
+    m_iconv = other.m_iconv;
+    m_targetSingleCharMaxLen = other.m_targetSingleCharMaxLen;
+
+    return *this;
+  }
+
+  std::string GetSourceCharset(void) const { return m_sourceCharset; }
+  std::string GetTargetCharset(void) const { return m_targetCharset; }
+  size_t GetTargetSingleCharMaxLen(void) const { return m_targetSingleCharMaxLen; }
+
+private:
+  std::string m_sourceCharset;
+  std::string m_targetCharset;
+  iconv_t m_iconv;
+  size_t m_targetSingleCharMaxLen;
+}; // CUnicodeConverterType
+
+CUnicodeConverterType::CUnicodeConverterType(const std::string_view sourceCharset,
+                                             const std::string_view targetCharset,
+                                             size_t targetSingleCharMaxLen /*= 1*/)
+  : m_sourceCharset(sourceCharset),
+    m_targetCharset(targetCharset),
+    m_iconv(NO_ICONV),
+    m_targetSingleCharMaxLen(targetSingleCharMaxLen)
+{
+}
+
+CUnicodeConverterType::~CUnicodeConverterType()
+{
+  if (m_iconv != NO_ICONV)
+  {
+    iconv_close(m_iconv);
+    m_iconv = NO_ICONV;
+  }
+}
+
+iconv_t CUnicodeConverterType::PrepareConverter()
+{
+  if (m_iconv == NO_ICONV)
+  {
+    m_iconv = iconv_open(m_targetCharset.data(), m_sourceCharset.data());
+
+    if (m_iconv == NO_ICONV)
+      CLog::Log(LOGERROR, "{}: iconv_open() for \"{}\" -> \"{}\" failed, errno = {} ({})",
+                __FUNCTION__, m_sourceCharset, m_targetCharset, errno, strerror(errno));
+  }
+  return m_iconv;
+}
+
+enum StdConversionType /* Keep it in sync with CUnicodeConverter::CInnerUCodeConverter::m_stdConversion */
+{
+  // Removed converters not needed for Unicode. It should not be difficult to re-add them.
+
+  NoConversion = -1,
+  Utf8ToUtf32 = 0,
+  Utf32ToUtf8,
+  Utf32ToW,
+  WToUtf32,
+  WToUtf8,
+  Utf8ToW,
+  NumberOfStdConversionTypes /* Dummy sentinel entry */
+}; // StdConversionType
+
+// This is where I went nuts creating templates. I am brand-new to Templates
```

Comment:
please keep in mind that your code is going into a large code base shared by many contributors and will probably live for many years, so it might make sense to save the comments about your personal skills for the PR description or commit message. In the code, they are a bit misplaced :)

Reply:
I wrote them for reviewers. Now is a good time to strip them. Next time I'll figure out how to annotate code using PR tools.